add symmetry information to DomainIntegral in Jintegral tests

BoZeng1997 opened the pull request:
add symmetry information to DomainIntegral in J integral tests so that the test results match the analytical value. Also updated the gold files. #24748

BoZeng1997 committed:
add symmetry information to DomainIntegral in Jintegral tests so that the test results match the analytical value. Also updated the gold files. #24748

recuero commented on modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_2d_mouth_dir_out.csv:
```diff
@@ -1,4 +1,3 @@
 time,J_1,J_2,J_3,J_4,J_5
 0,0,0,0,0,0
-1,1.1655970440759,1.1625839930871,1.1613679143305,1.164168518934,1.1632303584855
```
Why don't we get exactly double on this file?

BoZeng1997 replied:
I do not have an explanation. From the code in Jintegral.C
https://github.com/idaholab/moose/blob/51557c3cd1fd65ea9518d83a221f514524c39c99/modules/tensor_mechanics/src/vectorpostprocessors/JIntegral.C#L228-L229
the only place where the symmetry information takes effect is here, by doubling the j_integral result.
`modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_mouth_dir_out.csv` this one is the other result that does not give us exact doubling. They both use `crack_direction_method = CrackMouth` and default `crack_end_direction_method=NoSpecialTreatment`. The other tests with `crack_direction_method = CrackMouth` but `crack_end_direction_method=CrackDirectionVector` satisfy the double relationship.

recuero replied:
There is a piece of code that modifies the location of the crack front nodes here:

https://github.com/idaholab/moose/blob/51557c3cd1fd65ea9518d83a221f514524c39c99/modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C#L1063-L1066

I don't fully understand why we are zeroing the components of the symmetry plane for the entire crack mouth definition. It seems as if the nodes of the crack mouth could potentially move along the symmetry. If you change the symmetry plane to "2", you get exactly double the value (but it's obviously wrong in this case).

recuero replied:
@bwspenc: We can merge this PR once the tests pass, but I think we should address the reason why we don't get exactly double when we add the correct symmetry plane. I can open an issue if that cannot be address now.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1622215/) on 837e770 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24833/site)

This comment will be updated on new commits.

BoZeng1997 committed:
add symmtry information to test j_integral_2d_block_restrict.i
test j_integral_2d_mouth_dir.i and j_integral_3d_mouth_dir.i were commented out the symmetry condition because they do not return the expected exact doubled values. The case where we define the crack mouth need further check.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1686646/) on 1721185 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24833/coverage/xfem)

</details>



This comment will be updated on new commits.

recuero commented on modules/tensor_mechanics/test/tests/j_integral/gold/j_integral_3d_out.e:
Why do we need to regold this file?

BoZeng1997 replied:
the postprocessor values were written in exodus output. when postprocessor results are updated, all exodus results need to be regold.

recuero reviewed (approved):
Thanks for the contribution!
